makeSpectral.py

```diff
@@ -0,0 +1,259 @@
+from os import listdir, rename
+from os.path import isfile, join, abspath, basename, splitext, dirname
+from typing import override
+from typing import TextIO, List
+from shutil import copy, SameFileError
+from argparse import ArgumentParser
+
+"""
+NOTE:
+	Although a little more cumbersome, I feel that providing the our
+	*replace substring* function extra verbosity is important.
+	This is because the User's chosen Name and Description
+	may occur in parts of the c++ code we do not want to edit.
+	The inclusion of the context around the needle we want to substitute
+	will help prevent accidental substitions.
+Happy Coding. Pix ❤️
+""" 
+
+class Cmd:
+	def __init__(self, p : str, n : str, d : str, v : bool):
+		self.path = p.pop() if p else None
+		self.name = n.pop() if n else None
+		self.description = d.pop() if d else None
+		self.verbose = v.pop() if v else False
+
+class Jucer:
+
+	def __init__(self, name:str, desc:str, path:str):
+		self.name = name
+		self.desc = desc
+		self.path = path
+		self.diry = dirname(path)
+
+
+	def swit(self, haystack:str, needle:str, spore:str) -> str :
+		if haystack == "" or haystack is None:
+			return None
+		if not needle in haystack:
+			raise ValueError("needle not found in haystack")
+		return haystack.replace(needle, spore, 1)
+
+	def manip(self, dt : str) -> str:
```

Comment:
Ah yea, so this is a base class and the default method. To make that clearer, in the future we could use Python's abstract base class annotations:
https://docs.python.org/3/library/abc.html

But I don't think it's essential here

Reply:
Oh cool! I didn't know that library existed. Learned something 👍 
The `Jucer` object definitely has an abstract class feel to it and it's never called.
Happy to make that a future project. 😄

Reply:
Yea I've found it pretty useful in the past. Future project!